cli/replicasets/common.go

```diff
@@ -0,0 +1,185 @@
+package replicasets
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/context"
+	"github.com/tarantool/cartridge-cli/cli/project"
+	"github.com/tarantool/cartridge-cli/cli/running"
+	"gopkg.in/yaml.v2"
+)
+
+type InstanceConf struct {
+	URI string `yaml:"advertise_uri"`
+}
+
+type InstancesConf map[string]*InstanceConf
+
+func getInstancesConf(ctx *context.Ctx) (*InstancesConf, error) {
+	var err error
+
+	log.Debugf("Instances configuration file is %s", ctx.Running.RunDir)
+
+	if _, err := os.Stat(ctx.Running.ConfPath); err != nil {
+		return nil, fmt.Errorf("Failed to use instances configuration file: %s", err)
+	}
+
+	fileContentBytes, err := common.GetFileContentBytes(ctx.Running.ConfPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read instances configuration file: %s", err)
+	}
+
+	var allSectionsConf InstancesConf
+	if err := yaml.Unmarshal([]byte(fileContentBytes), &allSectionsConf); err != nil {
+		return nil, fmt.Errorf("Failed to parse instances configuration file %s: %s", ctx.Running.ConfPath, err)
+	}
+
+	instancesConf := make(InstancesConf)
+
+	appInstancePrefix := fmt.Sprintf("%s.", ctx.Project.Name)
+	for key, sectionConf := range allSectionsConf {
+		if strings.HasPrefix(key, appInstancePrefix) {
+			parts := strings.SplitN(key, ".", 2)
+			instanceName := parts[1]
+			instancesConf[instanceName] = sectionConf
+		}
+	}
+
+	return &instancesConf, nil
+}
+
+func getControlConn(instancesConf *InstancesConf, ctx *context.Ctx, joinInstances []string) (net.Conn, error) {
```

Comment:
Not sure `getControlConn` should have `joinInstances ` argument. Consider a better name for this function please

Reply:
This function is completely removed, see new `common` functions.